core/src/test/java/io/questdb/test/griffin/UpdateFuzzTest.java

```diff
@@ -0,0 +1,163 @@
+                insert(" insert into up " +
+                        " select timestamp_sequence('2022-02-25T03', 1000000) ts, '1' as x" +
+                        " from long_sequence(2)");
+
+            }
+
+            if (rnd.nextBoolean()) {
+                insert(" insert into up " +
+                        " select timestamp_sequence('2022-02-25T10', 1000000) ts, '1' as x" +
+                        " from long_sequence(2)");
+
+            }
+
+            if (rnd.nextBoolean()) {
+                insert(" insert into up " +
+                        " select timestamp_sequence('2022-02-26T10', 1000000) ts, '1' as x" +
+                        " from long_sequence(5)");
+            }
+
+            short lastType = ColumnType.STRING;
+
+            if (rnd.nextBoolean() && partitioned) {
+                insert(" insert into up " +
+                        " select timestamp_sequence('2022-02-24', 1000000) ts, '12' " +
+                        " from long_sequence(2)");
+            }
+
+            if (rnd.nextBoolean()) {
+                ddl("alter table up alter column x type symbol");
+                lastType = ColumnType.SYMBOL;
+                drainWalQueue();
+                try (ColumnPurgeJob purgeJob = new ColumnPurgeJob(engine)) {
+                    purgeJob.run(0);
+                }
+            }
+
+            if (rnd.nextBoolean()) {
+                update("UPDATE up SET x = '2' WHERE ts > '2022-02-25'");
+                drainWalQueue();
+                try (ColumnPurgeJob purgeJob = new ColumnPurgeJob(engine)) {
+                    purgeJob.run(0);
+                }
+            }
+
+            if (rnd.nextBoolean()) {
+                ddl("alter table up alter column x type varchar");
+                drainWalQueue();
+                lastType = ColumnType.VARCHAR;
+                try (ColumnPurgeJob purgeJob = new ColumnPurgeJob(engine)) {
+                    purgeJob.run(0);
+                }
+            }
+
+            if (rnd.nextBoolean() && partitioned) {
+                insert(" insert into up " +
+                        " select timestamp_sequence('2022-02-24T10', 1000000) ts, '1' as x" +
+                        " from long_sequence(1)");
+
+            }
+
+            if (rnd.nextBoolean() && lastType != ColumnType.STRING) {
+                ddl("alter table up alter column x type string");
+                drainWalQueue();
+                lastType = ColumnType.STRING;
+                try (ColumnPurgeJob purgeJob = new ColumnPurgeJob(engine)) {
+                    purgeJob.run(0);
+                }
+            }
+
+            if (rnd.nextBoolean() && lastType != ColumnType.SYMBOL) {
+                ddl("alter table up alter column x type symbol");
+                drainWalQueue();
+                try (ColumnPurgeJob purgeJob = new ColumnPurgeJob(engine)) {
+                    purgeJob.run(0);
+                }
+            }
+
+            printSql("up");
```

Comment:
nit: do we need this query execution considering that we don't do any later assertions on the sink?

Reply:
I added the comment 

```

            // This will read whole table, verifying that the TableReader can handle the column top / versions / types
            printSql("up");
```

Reply:
Thanks!